File: python/src/questao_9/main.py
def factorial(number:int):
   if(number <= 1):
    return 1
   
   return number * factorial(number - 1) 
   


def main():
   print("Digite um numero:")
   number = int(input())
   print("O fatorial de ",number, "é: ",factorial(number))
 

if __name__ == '__main__':
    main()

File: java/src/questao_9/Main.java
package questao_9;

import java.util.Scanner;

public class Main {
    
    public static int factorial(int number) {
        if (number <= 1) {
            return 1;
        }
        return number * factorial(number - 1);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Digite um numero:");
        int number = scanner.nextInt();
        scanner.close();
        System.out.println("O fatorial de " + number + " é: " + factorial(number));
    }
}

